File: years/2022/clojure/aoc-2022/test/aoc_2022/day_03_test.clj
(ns aoc-2022.day-03-test
  (:require [aoc-2022.day-03 :refer :all]
            [clojure.test :refer :all]))

(deftest day-03-test
  (is (= 157 (part1 "../../data/03-sample.txt")))
  (is (= 70 (part2 "../../data/03-sample.txt"))))

File: years/2024/elixir/test/aoc/day_03_test.exs
defmodule AoC.Day03.Test do
  @moduledoc false

  use ExUnit.Case, async: false

  # comment this out to always log to the console
  @moduletag :capture_log

  describe "part 1" do
    test "with sample data" do
      assert AoC.Day03.part_1([
               "xmul(2,4)%&mul[3,7]!@^do_not_mul(5,5)+mul(32,64]then(mul(11,8)mul(8,5))"
             ]) == 161
    end
  end

  describe "part 2" do
    test "with sample data" do
      assert AoC.Day03.part_2(["../data/03.txt"]) == 1
    end
  end
end
